app/dao/notifications_dao.py

```diff
@@ -508,40 +535,46 @@ def _delete_notifications(notification_type, date_to_delete_from, service_id, qu
     if isinstance(service_id, Row):
         service_id = str(service_id[0])
 
-    subquery = db.session.query(
-        Notification.id
-    ).join(NotificationHistory, NotificationHistory.id == Notification.id).filter(
-        Notification.notification_type == notification_type,
-        Notification.service_id == service_id,
-        Notification.created_at < date_to_delete_from,
-    ).limit(query_limit).subquery()
-
+    subquery = (
+        select(Notification.id)
+        .join(NotificationHistory, NotificationHistory.id == Notification.id)
+        .where(
+            Notification.notification_type == notification_type,
+            Notification.service_id == service_id,
+            Notification.created_at < date_to_delete_from,
+        )
+        .limit(query_limit)
+        .subquery()
+    )
     deleted = _delete_for_query(subquery)
 
-    subquery_for_test_keys = db.session.query(
-        Notification.id
-    ).filter(
-        Notification.notification_type == notification_type,
-        Notification.service_id == service_id,
-        Notification.created_at < date_to_delete_from,
-        Notification.key_type == KEY_TYPE_TEST
-    ).limit(query_limit).subquery()
+    subquery_for_test_keys = (
+        select(Notification.id)
+        .where(
+            Notification.notification_type == notification_type,
+            Notification.service_id == service_id,
+            Notification.created_at < date_to_delete_from,
+            Notification.key_type == KEY_TYPE_TEST
+        )
+        .limit(query_limit)
+        .subquery()
+    )
 
     deleted += _delete_for_query(subquery_for_test_keys)
-
     return deleted
 
 
 def _delete_for_query(subquery):
-    number_deleted = db.session.query(Notification).filter(
-        Notification.id.in_(subquery)).delete(synchronize_session='fetch')
-    deleted = number_deleted
+    stmt = delete(Notification).where(Notification.id.in_(subquery))
+    number_deleted = db.session.execute(stmt, execution_options={"synchronize_session": 'fetch'}).rowcount
```

Comment:
This is [legacy](https://docs.sqlalchemy.org/en/20/orm/queryguide/query.html#sqlalchemy.orm.Query.execution_options).  "execution_options" should be part of the query statement.  See the 2.0 example in inbound_sms_dao.py.

Reply:
Team agreed. The way to do it is to attach to .execute(stmt, execution_options={"synchronize_session": 'fetch'}). No change required.

Reply:
This is still valid.